perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java

```diff
@@ -1248,6 +1248,25 @@ public List<Attribute> getAttributes(PerunSession sess, User user, List<String>
 		}
 	}
 
+	@Override
+	public boolean isLoginAlreadyUsed(PerunSession sess, String login, String namespace) {
+		try {
+			if (namespace == null) {
+				return jdbc.queryForInt("select count(*) from attr_names as attr join user_attr_values attr_val on attr.id=attr_val.attr_id\n" +
+					"where attr.friendly_name like 'login-namespace:%' \n" +
+					"    and attr.friendly_name not like '%persistent%' \n" +
+					"    and attr_val.attr_value like ?", login) > 0;
+			} else {
+				return jdbc.queryForInt(String.format("select count(*) from attr_names as attr join user_attr_values attr_val on attr.id=attr_val.attr_id\n" +
+					"where attr.friendly_name like 'login-namespace:%s' \n" +
+					"    and attr.friendly_name not like '%%persistent%%' \n" +
+					"    and attr_val.attr_value like ?", namespace), login) > 0;
+			}
```

Comment:
Could we just set the namespace to empty string if it is null and keep the else clause ? Since right now both queries are almost identical.

Reply:
Good point, I just need to replace it with `%` instead of an empty string if the namespace is NULL, but yes, now it is more readable, thanks.